### DISCUSSION TITLE
Sudden, Silent License change?

### DISCUSSION THREAD

**Discussion by dsignr** [minio/minio#12157] [Q&A] [answered]
It's perfectly fine to change the license, but when tons of people are using your product, this may cause legal woes for many especially if the new license isn't compatible. Normally license change are announced in some form of PR or announcement or discussion and none of that has taken place. Hence, creating this issue for others to notice.

I personally don't have an issue with the license, but I find this silent change unethical and slimy.

**Comment by harshavardhana** [comment, marked as answer]
> It's perfectly fine to change the license, but when tons of people are using your product, this may cause legal woes for many especially if the new license isn't compatible. Normally license change are announced in some form of PR or announcement or discussion and none of that has taken place. Hence, creating this issue for others to notice.

Almost all our other projects are already in AGPLv3 since 2019 - not sure why this should be surprising @dsignr - our website also has always talked about AGPLv3 for some time now https://min.io - this is just a natural progression for us. 

The change is done here in the source first, we will be making an announcement shortly.

**Comment by pdf** [comment]
Unless your contributors have signed a CLA allowing you to change the license for their contributions, you may have legal problems here (IANAL).

**Comment by ddevault** [comment]
You can sublicense Apache 2.0 projects and the Apache license is GPL-compatible - so they should be legally in the clear. Though IANAL, either. It doesn't work backwards, though - MinIO is hereby forever committed to ship this codebase as AGPL.

It's pretty frustrating to have it sprung on everyone without notice or community input, though. MinIO has a problem with not giving its community adequate care or attention. It's MinIO's project and we're just using it.

**Comment by erir910** [comment]
I love how people who almost never contribute to open source are the most outraged when it comes to re-licensing and are the ones who lead the crusade smearing open source authors and startups. MinIO was always AGPL. Almost all contributions are done by MiniIO employees. Stop this arrogant sef-entitlement.

**Comment by ddevault** [comment]
The AGPL is a great license, I use it on many of my own projects (I work full time on open source, by the way). That has nothing to do with whether or not this is a nice way to treat your community.

>MinIO was always AGPL.

It was Apache 2.0 before.

Tamper down on the flame war, okay?

**Comment by erir910** [comment]
MinIO has always been AGPLv3 and their business model was based on dual commercial licensing and support. Also they literally just added the relicinsing commits a couple of hours ago. It's not like they did it 3 months ago and you found out by chance. It's not like they relicensed it to a proprietary license either. Not to mention that almost all users don't have to care whether minio is apache or agpl anyway.

**Comment by ddevault** [comment]
I'm not sure what your crusade is for, but there's a lot of factual errors in your claims. I can't tell you what their business model is or was, but I can tell you that Minio-the-software was licensed Apache 2.0 before. That's a fact. Their other software might have been, but this was not their other software. I'm not sure what the timing has to do with anything, of course we're talking about it when it's fresh.

I am grateful that they did not relicense it to a proprietary license. There are degrees of good and bad conduct. Changing to a nonfree license would be a severe party foul, but springing a change on everyone, unannounced and without community involvement, is also not a good look.

If users didn't care what the license was, then we would not go to the trouble of sitting down to talk about it here.

Please remember that we're all discussing the matter in good faith and refrain from making inflammatory arguments.

**Comment by jchv** [comment]
I concur that announcement-first would’ve been better. I realize this thread is about to become a lot of “me too”s but that’s the most important takeaway for me.

**Comment by dsignr** [comment]
> I love how people who almost never contribute to open source are the most outraged when it comes to re-licensing and are the ones who lead the crusade smearing open source authors and startups. MinIO was always AGPL. Almost all contributions are done by MiniIO employees. Stop this arrogant sef-entitlement.

First off, the validity of my point is not affected by my contribution towards Open Source. Also, for the record, I maintain [open source repositories](https://github.com/dsignr/disallowed-usernames) too, but I would never silently change the license hoping that no one would notice.

Next, the original license was different than what it is now, so please go back and kindly check as to why this issue exists in the first place.
This literally affects people in companies where AGPL is frowned upon, and this may cause issues for their day to day work if it isn't communicated properly. I have been in this situation before. Try explaining to legal about this situation, it will always fall on you, the developer, and it would look as though you didn't do your due diligence.

This is NOT "sef-entitlement" for pointing out the above in good faith. I literally do not benefit from any of this. Tone down a bit perhaps and we can continue a constructive discussion.

**Comment by ddevault** [comment]
It's also worth noting that software licenses *do* entitle you to certain rights, so entitlement is often entirely appropriate in discussions like these. Contributors to this project are *legally* entitled to the conditions of the Apache 2.0 license which their changes were provided under.

**Comment by erir910** [comment]
> No offence..but
> 
> > I love how people who almost never contribute to open source...
> 
> ![image](https://user-images.githubusercontent.com/5097028/115941724-40a48080-a4c4-11eb-9ef5-3fd98cee09b1.png)

Have you ever wondered that this might be some sort of a throwaway account?

**Comment by ddevault** [comment]
Have you wondered if the people who you believe "almost never contribute to open source" could be using a throwaway account? If you want to have people check in their FOSS creed at the door to these discussions, start with yourself.

**Comment by erir910** [comment]
@ddevault I am not sure why you want to pick up a fight with me and I don't think I have to list my popular foss projects just to justify my opinions. I know that writing inflammatory blog posts and picking up fights with people online is your main talent but I am not falling for that.

**Comment by ddevault** [comment]
> don't think I have to list my popular foss projects just to justify my opinions.

"Rules for thee, but not for me"?

>I love how people who almost never contribute to open source are the most outraged when it comes to re-licensing and are the ones who lead the crusade smearing open source authors and startups.

You picked the fight. Everyone here was pretty cordial until you showed up. Knock it off, dude.

**Comment by pdf** [comment]
> You can sublicense Apache 2.0 projects and the Apache license is GPL-compatible

Contributions made under Apache 2.0 will remain under that license though, right? So they can't just relicense without at least acknowledging that.

Regardless, that's for lawyers to work out. Handling of license changes must involve significant care, and as mentioned elsewhere, this silent change will have an impact for those working in companies that have defective policies related to deployment of AGPL code.

**Comment by ddevault** [comment]
>Contributions made under Apache 2.0 will remain under that license though, right? So they can't just relicense without at least acknowledging that.

Well, kind of. I can see some interpretation here with respect to the specific relationship between the two licenses, the codebase, and the copyright owners. This one is for the lawyers to argue about. If I were considering such a change I would have taken more pains to avoid an ambiguous situation here.

The MinIO team clearly thinks of the project as *theirs* and the contributions of others as, at best, a minor inconvenience. We can see this in how they failed to consult with the community on this change, or provide notice to anyone before making it. I got that feeling from prior interactions with them as well. Because they view the code as *theirs* (exclusive of the community), not *ours* (inclusive of the community), they don't seem to make much of an effort to respect & value the contributions of their community.

**Comment by erir910** [comment]
I didn't pick a fight with anybody. I am just tired of this sense of self-entitlement whenever a popular foss project is relicensed (even to SSPL or BSL like it happened many times lately). The smearing campaign always starts on HN a couple of minutes later after the relicensing and people talk about how this is illegal, CLAs about mostly very trivial commits, backstabbing as if they are entitled to anything in the first place, etc...

MinIO is open source company, their product MinIO was always AGPLv3. MiniIO unlike many commercial open source projects was never made open-core or gaslighting people with open source like many others (looking at you tailscale and others who use foss-client locked in third party server model). I am not sure who is benefiting from smearing open source/source-available companies but I am sure if this continues we will never see high quality complex open source software (e.g. minio, cockroachdb, etc...) that is free of baiting and gaslighting

**Comment by thayolee** [comment]
@erir910 At this point your identity is obvious dude. @harshavardhana could you please at this point continue conversations from your real account instead of trolling from the other account? No need to get passive aggressive over this hiding behind a fake ID dude.

No one else has the incentive to get passive aggressive over this, plus your commits in the troll account tie back to your real account (go figure which one..lol)

**Comment by dsignr** [comment]
@harshavardhana I understand maintaining open source is tough, and mistakes like these are bound to happen every now and then. I do not wish to criticize nor diminish your contributions to open source and this project in general. I hope you take my issue as constructive criticism. I apologize for the direction in which this thread is going, I will refrain from replying if it helps. I am unsubscribing for my own sanity and sorry this is going way off track than I originally intended it for. Consider locking the thread if it helps. As long as you got the point, I think we should be good.

Thank you for your contributions.
Cheers.

**Comment by harshavardhana** [comment]
> @erir910 At this point your identity is obvious dude. @harshavardhana could you please at this point continue conversations from your real account instead of trolling from the other account? No need to get passive aggressive over this hiding behind a fake ID dude.
>
 
People are referring to this thread and wrongly judging us without doing any research. I previously closed the thread upon author's request.

![linkedin](https://user-images.githubusercontent.com/622699/115969176-86773e00-a4f0-11eb-86c5-0271681072ad.png)

My account handle is @harshavardhana and I have no idea who @erir910 is.

This change is not silent. We will certainly make a release in the next few days and provide a detailed FAQ clarifying this change. Since we do not have a private repo for MinIO and our upsteam is public, we have to merge these changes before the release. Community was quick to notice these commits. Please be patient. We are just 2 to 3 days away from our next public release and it will have a detailed FAQ about his license change.
